engine/src/main/java/org/terasology/rendering/dag/stateChanges/BindFBO.java

```diff
@@ -37,10 +37,10 @@
 
     public BindFBO(String fboName) {
         this.fboName = fboName;
-        if (fboName.equals(DEFAULT_FRAME_BUFFER_NAME)) {
-            fboId = DEFAULT_FRAME_BUFFER_ID;
-        } else {
+        if (!fboName.equals(DEFAULT_FRAME_BUFFER_NAME)) {
             fboId = frameBuffersManager.getFBO(fboName).fboId;
+        } else {
+            fboId = DEFAULT_FRAME_BUFFER_ID;
```

Comment:
Sorry. I was hoping this PR could get through but I still it has some issues to deal with. 

In line with what we discussed regarding SetViewportSizeOf here it makes sense to just store the fboName and not worry about the fboId. The fboId is not needed yet AND if it was needed we'd have to subscribe to the FBM right away.

So, I'd remove the IF/ELSE block completely. The fboId can be fetched within the update() method and the generateTask() method. I'd also remove the fboId variable completely: let the BindFBOTask class print it out in its toString() method.

Finally, please review the generateTask method. it can check if `fboName != DEFAULT_FRAME_BUFFER_NAME` instead of checking the id. There's a small performance loss (integer vs string comparison) but it's not hot code so I don't think we shouldn't worry about it. You might also want to add an uninitialized static variable defaultTask. In generateTask() it would get filled by an unchangeable default task.